RPS.js

```diff
@@ -15,127 +15,86 @@ function getCpuChoice () {
     return cpuChoice;
 }
 
-const rockBtn = document.getElementById('rock');
-rockBtn.addEventListener('click', rockChoice);
-function rockChoice () {
-    console.log('The function rockChoice() is running....');
-    cpuButtonHighlightRemoval(cpuRock);
-    cpuButtonHighlightRemoval(cpuPaper);
-    cpuButtonHighlightRemoval(cpuScissors);
-    let winnerOfRound = '';
-    let cpuChoice = getCpuChoice();
-    let playerChoice = 'rock';
-    if (playerChoice === cpuChoice) {
-        winnerOfRound = 'Draw, there is no winner';
-        console.log(winnerOfRound);
-        cpuButtonHighlight(cpuRock);
-        playerCPUDraw(playerScore, cpuScore); 
-        return changeBannerText(winnerOfRound);
-    }
-    else if (playerChoice == 'rock' && cpuChoice === 'paper') {
-        winnerOfRound = 'You lose! Paper beats rock!';
-        console.log(winnerOfRound);
-        cpuButtonHighlight(cpuPaper);
-        cpuIncrementScore(cpuScore, cpuScoreElement);
-        return changeBannerText(winnerOfRound);
-    }
-    else if (playerChoice == 'rock' && cpuChoice === 'scissors') {
-        winnerOfRound = 'You win! Rock beats scissors!';
-        console.log(winnerOfRound);
-        cpuButtonHighlight(cpuScissors);
-        playerIncrementScore(playerScore, playerScoreElement);
-        return changeBannerText(winnerOfRound);
+
+// 0 == tie, 1 == victory, -1 == loss
+const resultMatrix = [
+                                // CPU
+                        // rock, paper, scissors
+    /*Player   rock     */[ 0,     -1,      1],
+    /*         paper    */[ 1,      0,     -1],
+    /*         scissors */[-1,      1,      0]
+
+]
```

Comment:
Pros:
We have a central place to modify results if we were to extend this. What if we put a spin on this game and reverse the outcomes? So paper > scissors > rock > paper. We could change the values here instead of modifying all the `if/else`. Also what if we wanted to add a new choice, say 'sword', that beats scissors and paper. We could easily change the values here, but having a separate handler with `if/else` would result in exponentially more branches to write out.

Cons:
Pretty difficult to read. Converting the button click to it's value and then a numerical choice for the indices (lines 30-34 below) has a lot of indirection that can be confusing.